core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -539,6 +539,7 @@ oppia.directive('conversationSkin', [
                         hasContinueButton: false
                       });
                     }
+                    $scope.$broadcast('stayOnSameCard');
```

Comment:
I don't think I understand what's going on here. We try to avoid $broadcast because it turns the code into spaghetti (it's like using goto). But I don't understand why we're doing this in order to cancel a timeout. Shouldn't we just set up the timeout only after Oppia's response comes back to the client?

Or, to put it another way, is there a way to refactor this so that you don't need the $broadcast?